jnosql-mapping/jnosql-mapping-core/src/main/java/org/eclipse/jnosql/mapping/core/NoSQLPage.java

```diff
@@ -64,6 +64,16 @@ public int numberOfElements() {
         return this.entities.size();
     }
 
+    @Override
+    public boolean hasNext() {
+        throw new UnsupportedOperationException("Eclipse JNoSQL has no support for this feature ");
```

Comment:
Maybe it would be interesting to also add which feature is not supported along with the error message
```suggestion
        throw new UnsupportedOperationException(new StringBuilder("Eclipse JNoSQL has no support for this feature ").append("hasNext").toString());
```